src/main/java/Chatterbox.java

```diff
@@ -0,0 +1,125 @@
+import chatterboxerrors.ChatterBoxDataFileError;
+import chatterboxerrors.ChatterBoxError;
+import chatterboxerrors.ChatterBoxNullTaskError;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.ToDo;
+import utils.Commands;
+import utils.Parser;
+import utils.Storage;
+import utils.StoredList;
+import utils.Ui;
+
+
+/**
+ * Represents a Chatbot.
+ */
+public class Chatterbox {
+    private final StoredList taskList;
+    private final Storage storage;
+    private final Ui ui;
+
+    /**
+     * Initialised an instance of Chatterbox with the given save file.
+     * @param saveFilePath The filepath of a save file.
+     */
+    public Chatterbox(String saveFilePath) {
+        storage = new Storage(saveFilePath);
+        ui = new Ui();
+        try {
+            storage.readFromSave();
+        } catch (ChatterBoxDataFileError e) {
+            ui.printMessage(e.getMessage());
+        }
+        taskList = storage.getSaveList();
+        ui.printWelcome();
+        ui.printTasks(taskList);
+    }
+
+    /**
+     * Runs Chatterbox until a bye input is provided by the user.
+     */
+    public void run() {
+        boolean isRunningProgram = true;
+        while (isRunningProgram) {
+            String userInput = ui.getUserInput();
+            try {
+                isRunningProgram = doCommand(userInput);
+            } catch (ChatterBoxError e) {
+                ui.printMessage(e.getMessage());
+            }
+        }
+    }
+
+    /**
```

Comment:
All of the Javadoc comments are clear and well-written, making the code easy to understand and maintain 🎉